backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -350,7 +353,22 @@ export class FeatureFlagService {
         updatedAt,
         ...flagDoc
       } = flag;
+
+      if (oldFlagDoc.context !== flagDoc.context) {
+        includeList?.forEach((list) => {
+          this.deleteList(list.segment.id, FEATURE_FLAG_LIST_FILTER_MODE.INCLUSION, user, logger);
+        });
+
+        excludeList?.forEach((list) => {
+          this.deleteList(list.segment.id, FEATURE_FLAG_LIST_FILTER_MODE.EXCLUSION, user, logger);
+        });
```

Comment:
This should be the part of the transaction and should be deleted in a single database call. Please use await.